docs/signature-layout.md

```diff
@@ -0,0 +1,35 @@
+# Signature File Layout
+
+A common file layout for storing and serving signatures provides a consistent way to reference image signatures. Signatures on a filesystem or a web server shall use this common layout. Signatures stored in a REST API are not required to use this common layout.
+
+## Specification
+
+This specification relies on [RFC3986](https://tools.ietf.org/html/rfc3986), focusing on defining a [path component](https://tools.ietf.org/html/rfc3986#section-3.3) to compose a concise URI reference to a signature.
+
+**SCHEME[AUTHORITY]/PATH_PREFIX/IMAGE@MANIFEST_DIGEST/signature-INT**
+
+**Definitions**
+
+* **SCHEME**: URI scheme per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.1), e.g. **file://** or **https://**
+* **AUTHORITY**: An optional authority reference per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.2), e.g. **example.com**
+* **PATH_PREFIX**: An arbitrary base path to the image component
+* **IMAGE**: The name of the image per [v2 API](https://docs.docker.com/registry/spec/api/#/overview). This would typically take the form of registry/repository/image but is not required to have exactly three parts. There is no requirement to include a **:PORT** component but it should be included if part of the image reference.
```

Comment:
>> The problem not solved by the namespaced-sigstore model, as this PR defines it, is one of mirroring. Consider that I have an on-premises registry at reg.hrivnak.org. I use it to have a local mirror of third-part content, so it includes repositories like redhat/rhel7. A local client tries to pull reg.hrivnak.org/redhat/rhel7:latest and gets some content. At what URL should it expect to find a signature?

> An unconfigured client would not have a sigstore configuration for reg.hrivnak.org and it will not fetch any signatures.
>
> Again, the conceptual model is that image+signatures are a single unit always hosted together. So, if you have a docker/distribution registry mirror, you should also have a corresponding sigstore mirror, and the clients need to be configured to know the sigstore location for reg-sigstore.hrivnak.org.

I see your point and agree, but there is a still a gap. The spec defined by this PR requires that the client insert the hostname of the original Red Hat registry into the URL. The client knows nothing about that registry, right?. Unless the intent is that all signatures on ``reg-sigstore.hrivnak.org`` would be namespaced to ``reg.hrivnak.org``? But if that's the case, why include a hostname at all?

> there would need to be a single world-wide sigstore. 

No, not at all!

The idea is that any individual sigstore could host whatever collection of signatures are valuable to its audience of users. If you are an authority, you can host a sigstore with your signatures. If you want to mirror one or more authorities, you mirror the signature files from those authoritative sigstores. There's nothing centralized about it.


>> @mtrmac I have a feeling that you have a particular scenario in mind that involves many users, many keys, and lots of access restrictions. Perhaps it would help to define those use cases directly and explicitly, and then we could more easily see which designs would accommodate them.

> I think the conceptual hurdle is to ignore how the sigstore technically works and to see that containers/image generally treats manifest+layers+signatures as single atomic unit. Then the philosophy and access control issues etc. follow naturally.

The reason for this PR is to define aspects of how the sigstore technically works. I don't know what to do with your suggestion to ignore that.

I see your general point about how this library solves a lot of problems, and that's useful. But I'm not sure that "just use my library and write your code in the language I chose" will be sufficient. I genuinely think this discussion will be easier if we describe all the use cases we're trying to solve, and then try to define a data model and interface that matches. We're coming from different perspectives with different assumptions, so let's define them clearly. I want to better understand the use cases you have in mind so I can better understand how mirroring fits in.

> (FWIW note that naive mirroring signatures with names cannot “just work” because the signatures claim the original (redhat/rhel7) identities, so pulling the reg.hrivnak.org/redhat/rhel7:latest will see an identity mismatch. That is quite unrelated to the sigstore mechanism, and we are prepared to support that, see hostname:5000/vendor/product in https://github.com/containers/image/blob/master/docs/policy.json.md#examples .)

This is very interesting. Thanks.

But per https://github.com/containers/image/pull/59/files#diff-c44796aa0013711d899f7345d01b8185R37, the signature would only include the name of the repository, and not include a registry hostname. I'm not sure where the identity mismatch would happen. Or is that PR (which is still not merged as of this writing) not accurate?